startup/gui/menus.py

```diff
@@ -164,34 +164,7 @@ def addHelpMenuItems( items ) :
 		import GafferDelight
 		import GafferDelightUI
 
-		def __shaderNodeCreator( nodeName, shaderName ) :
-
-			node = GafferOSL.OSLShader( nodeName )
-			node.loadShader( "maya/osl/" + shaderName )
-
-			return node
-
-		for label, shader in [
-			( "Standard", "material3Delight" ),
-			( "Glass", "material3DelightGlass" ),
-			( "Metal", "material3DelightMetal" ),
-			( "Skin", "material3DelightSkin" ),
-			( "Hair", "materialHairAndFur" ),
-		] :
-
-			nodeMenu.append(
-				"/3Delight/Shader/" + label,
-				functools.partial( __shaderNodeCreator, label, shader ),
-				searchText = "dl" + label
-			)
-
-		GafferSceneUI.ShaderUI.appendShaders(
-			nodeMenu.definition(), "/3Delight/Shader/Maya",
-			[ os.path.join( os.environ["DELIGHT"], "maya", "osl" ) ],
-			[ "oso" ],
-			__shaderNodeCreator,
-			searchTextPrefix = "maya"
-		)
+		GafferDelightUI.ShaderMenu.appendShaders( nodeMenu.definition() )
```

Comment:
It looks like this will mean we parse all the shaders with OSLQuery before the menu is even shown, which could impact startup times. Is that right, and have you measured the impact at all? I wonder if we could use the `subMenu` callable feature of MenuDefinition so we only do the parsing when the particular submenu is opened?

Reply:
I didn't measure the performance impact, but I didn't notice much difference when launching. But since the capability is there, I may as well use it. I now defer the loading until menu time in https://github.com/GafferHQ/gaffer/pull/5459/commits/2ff04ae3607033a8c0b756e1b9a314d64f2bea65.

I also make a simple cache of the menu items in that commit. Is that worth doing or are the menu definitions cached elsewhere?